### PR TITLE
HIP-103 phase 1

### PR DESCRIPTION
**Pull request by maplant**
I think this needs a bit of a re-work to address how we use disktree, but it might just end up being fine as-is

### CODE REVIEW

File: mobile_verifier/src/coverage.rs [hunk 1]
```diff
@@ -568,61 +571,137 @@ impl CoveredHexes {
...
+        let urbanization = urbanization_cache.clone();
         let outdoor_rewards = self.outdoor.into_iter().flat_map(move |(hex, radios)| {
+            let urbanization = urbanization.clone();
             radios
                 .into_sorted_vec()
                 .into_iter()
                 .take(MAX_OUTDOOR_RADIOS_PER_RES12_HEX)
                 .zip(OUTDOOR_REWARD_WEIGHTS)
                 .map(move |(cl, rank)| {
+                    let oracle_multiplier = if urbanization.get(&hex)? {
```
[the remark this answers is not in the document]
Comment (maplant) [reply mobile_verifier/src/coverage.rs]: The iterator returns a Result

File: mobile_verifier/src/settings.rs [hunk 2]
```diff
@@ -42,6 +42,7 @@ pub struct Settings {
     pub cbrs_geofence_regions: String,
     #[serde(default = "default_fencing_resolution")]
     pub cbrs_fencing_resolution: u8,
+    pub urbanization_data_set: PathBuf,
```
[the remark this answers is not in the document]
Comment (maplant) [reply mobile_verifier/src/settings.rs:45]: it wouldn't compile if it didn't

**Review by andymck — APPROVED** [review]

**Comment by maplant** [comment]
> how much extra work is it to convert a file of raw integers (perhaps compressed) into a DiskTreeMap on boot or how much more size on disk is the raw hex integer file? is there any value in keeping the urbanization input file in a human-readable format and doing a one-time conversion at start to DiskTreeMap to avoid the need to pull in a non-release version of HexTree and keep the urbanization data a less inscrutable bincode format?

I have no idea. I don't know what the disk tree map format is at all

File: mobile_verifier/src/coverage.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (maplant) [reply mobile_verifier/src/coverage.rs]: I'm really not sure what you're saying

[the remark this answers is not in the document]
Comment (maplant) [reply mobile_verifier/src/coverage.rs]: It doesn't. It returns an `Ok(bool)`. Take a look at the implementation of `DiskTreeCacher`

**Pull request closed without merging**